### PR TITLE
🐛 Skip bundle defaults if --tag passed

### LINKED ISSUE #1180: Porter parses local porter.yaml even when using tag flag
## Describe the bug
If there is a `porter.yaml` file in the current working directory, Porter systematically parses it even if one is trying to use a bundle directly from a registry.

## To Reproduce
```console
$ mkdir empty && cd empty
$ touch porter.yaml
$ porter install --tag deislabs/porter-hello-bundle:latest
Error: invalid tag : invalid reference format

$ rm porter.yaml
$ porter install --tag deislabs/porter-hello-bundle:latest
installing HELLO...
executing install action from HELLO (bundle instance: HELLO)
Install Hello World
Hello World
execution completed successfully!
```

## Expected behavior
I would expect that when doing an operation with `--tag` that the local `porter.yaml` would be ignored.

## Porter Command and Output
See steps to reproduce

## Version
`porter v0.27.2 (aee93e98)`

### PR DESCRIPTION
**Pull request by dev-drprasad**
# What does this change
if `--tag` passed, porter still tries to validate local manifest file. This PR skips validating manifest file if `--tag` present

# What issue does it fix
Closes #1180

[1]: /CONTRIBUTING.md#when-to-open-a-pull-request

# Notes for the reviewer
N/A

# Checklist
- [x] Unit Tests
- [x] Documentation - N/A
- [x] Schema (porter.yaml) - N/A

If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

[contributors]: /CONTRIBUTORS.md

### CODE REVIEW

**Comment by dev-drprasad** [comment]
In draft, cuz thinking of adding unit tests

**Comment by dev-drprasad** [comment]
@vdice I feel, your solution directly convey the problem we're trying to solve. Let incorporate those changes only

File: pkg/porter/lifecycle_test.go [hunk 1]
```diff
@@ -158,6 +160,18 @@ func TestBundleLifecycleOpts_ToActionArgs(t *testing.T) {
 		assert.Equal(t, opts.Name, args.Installation, "Claim not populated correctly")
 		assert.Equal(t, opts.RelocationMapping, args.RelocationMapping, "RelocationMapping not populated correctly")
 	})
+
+	t.Run("ignore manifest in cwd if tag present", func(t *testing.T) {
```
Comment (vdice) [inline pkg/porter/lifecycle_test.go:164]: Surprisingly, I couldn't get this test to fail when running against the current code on `main` (meaning, without the changes to cnab.go and lifecycle.go above).  Maybe double-check my work.

I succeeded in getting it to fail when explicitly specifying `opts.File = "porter.yaml"` (and it didn't seem like the working dir logic was strictly necessary):

```
+++ b/pkg/porter/lifecycle_test.go
@@ -1,8 +1,6 @@
 package porter

 import (
-       "os"
-       "path"
        "path/filepath"
        "testing"

@@ -164,13 +162,11 @@ func TestBundleLifecycleOpts_ToActionArgs(t *testing.T) {
        t.Run("ignore manifest in cwd if tag present", func(t *testing.T) {
                opts := BundleLifecycleOpts{}
                opts.Tag = "deislabs/kubekahn:latest"
+               opts.File = "porter.yaml"

-               // cnab.go#defaultBundleFiles uses `os.Getwd()`
-               wd, _ := os.Getwd()
-               p.TestConfig.TestContext.AddTestFileContents([]byte(""), path.Join(wd, "porter.yaml"))
+               p.TestConfig.TestContext.AddTestFileContents([]byte(""), "porter.yaml")
                err := opts.Validate(nil, p.Porter)
                require.NoError(t, err, "Validate failed")
-               p.TestConfig.TestContext.FileSystem.Remove(path.Join(wd, "porter.yaml"))
        })
 }
```

We don't necessarily need to go with the test mods above, but we should make sure that this test fails without the actual code changes elsewhere in this PR.

Comment (dev-drprasad) [reply pkg/porter/lifecycle_test.go:167]: > Surprisingly, I couldn't get this test to fail when running against the current code on main

My bad! Thanks for catching this. Test case indeed failing in my local with `main` branch. But the reason was something else and I thought test case working as expected 🤦🏻‍♂️

`opts.File = ''porter.yaml"` simulates explicitly passing of `--file` arg. I wanted to test default behaviour without passing `--file` arg

File: pkg/porter/lifecycle_test.go [hunk 2]
```diff
@@ -160,6 +162,25 @@ func TestBundleLifecycleOpts_ToActionArgs(t *testing.T) {
...
+		opts := BundleLifecycleOpts{}
+		opts.Tag = "deislabs/kubekahn:latest"
+
+		wd, _ := os.Getwd()
+		// Below line makes cnab.go#defaultBundleFiles#manifestExists true
+		p.TestConfig.TestContext.AddTestFileContents([]byte(""), path.Join(wd, config.Name))
+		// When execution reach to `readFromFile`, manifest file path will be lost.
+		// So, had to empty default manifest content also
+		// Below line makes manifest.go#readFromFile#exists true
+		p.TestConfig.TestContext.AddTestFileContents([]byte(""), config.Name)
```
Comment (dev-drprasad) [inline pkg/porter/lifecycle_test.go:177]: I couldn't make this test fail with `main` branch with just one empty manifest file. 

`defaultBundleFiles#manifestExists` will be `true` only if I use `path.Join(wd....` and we need it to execute bundle default logic.
But `readFromFile` func gets only filename (not complete path) and reads  manifest file from root directory. So, I had to use root manifest file also

Comment (vdice) [reply pkg/porter/lifecycle_test.go]: Ah I see.  The in-line comments help!  Thanks for updating.

**Pull request merged**